rust/connlib/tunnel/src/peer.rs

```diff
@@ -273,7 +273,7 @@ impl ClientOnGateway {
         let expired_translations = self
             .permanent_translations
             .iter()
-            .filter(|(_, state)| state.no_response_in_30s(now));
+            .filter(|(_, state)| state.no_response_in_30s(now) && state.is_used(now));
```

Comment:
Did you purposely not add the same condition on line 290?

I guess we want to refresh a domain as long as one IP is used but they are all stale and not just if they are all still being used.

Reply:
yes, precisely